### PR TITLE
NET-39 nm-quick.sh refactor (SSL certificates)

### PR DESCRIPTION
**Pull request by TobiaszCudnik**
## Describe your changes

NET-39 - `nm-quick.sh` refactor

Changes:
- added `nm-certs.sh` encapsulating all certificate weightlifting
  - via [certbot](https://github.com/certbot) and [zerossl-certbot](https://github.com/zerossl/zerossl-bot)
- added `netmaker.env` config (email, domain for now)
- code cleanups
- basic support for relative paths
- crontab file to renew the certificates (monthly)

Temporarily based off https://github.com/gravitl/netmaker/pull/2274 to reduce the diff size.

## Provide testing steps

```
wget https://raw.githubusercontent.com/gravitl/netmaker/NET-39/nm-quick-refactor/scripts/nm-quick.sh
chmod +x nm-quick.sh
./nm-quick.sh -b local -t NET-39/nm-quick-refactor
```

Notes / scenarios:
1. Repeat the process to assert cert rate limiting doesnt happen
3. Both `nip.io` and a custom domain
4. Assert `netmaker.env` gets created (same dir as `nm-quick.sh`)
5. Assert `/etc/cron.monthly/nm-certs.sh` exists and is runable

## Checklist before requesting a review
- [x] My changes affect only 10 files or less.
- [x] I have performed a self-review of my code and tested it.
- [x] If it is a new feature, I have added thorough tests, my code is <= 1450 lines.
- [x] If it is a bugfix, my code is <= 200 lines.
- [x] My functions are <= 80 lines.
- [ ] I have had my code reviewed by a peer.
- [x] My unit tests pass locally.
- [x] Netmaker is awesome.

### CODE REVIEW

File: scripts/nm-quick.sh [hunk 1]
```diff
@@ -281,49 +284,46 @@ install_dependencies() {
 
 	OS=$(uname)
 	if [ -f /etc/debian_version ]; then
-		dependencies="git wireguard wireguard-tools dnsutils jq docker.io docker-compose"
+		dependencies="git wireguard wireguard-tools dnsutils jq docker.io certbot docker-compose python-is-python3"
```
Comment (mattkasun) [inline scripts/nm-quick.sh:287]: python only required on debian?

Comment (TobiaszCudnik) [reply scripts/nm-quick.sh:287]: Debian / Ubuntu have `python2` & `python3`. `python-is-python3` is just a symlink pkg.

It's needed for the ZeroSSL certbot wrapper, while the regular python package is a dep of certbot.

Comment (TobiaszCudnik) [reply scripts/nm-quick.sh:287]: Just check Alpine and it also causes problems. Will use certbot via docker instead.

**Review by afeiszli — APPROVED** [review]

**Comment by theguy951357** [comment]
current github issues are keeping the script from checking out branches. will test again when github is working again

**Review by theguy951357 — CHANGES_REQUESTED** [review]

File: scripts/nm-quick.sh [hunk 2]
```diff
@@ -246,18 +231,36 @@ confirm() { (
...
+	fi
+)}
+
 # local_install_setup - builds artifacts based on specified branch locally to use in install
 local_install_setup() { (
 	rm -rf netmaker-tmp
 	mkdir netmaker-tmp
 	cd netmaker-tmp
-	git clone https://www.github.com/gravitl/netmaker
+	git clone --depth=1 https://www.github.com/gravitl/netmaker
```
[the remark this answers is not in the document]
Comment (TobiaszCudnik) [reply scripts/nm-quick.sh]: Fixed.

**Review by theguy951357 — APPROVED** [review]
netmaker installs. certificates good.
critical list passes

**Pull request merged**